src/Agent.Worker/TestResults/ResultsCommandExtension.cs

```diff
@@ -289,8 +288,10 @@ private async Task PublishTestRunDataAsync(String teamProject, TestRunContext te
             }
 
             await PublishEventsAsync(connection);
+            TestResultsHttpClient tcmClient = connection.GetClient<TestResultsHttpClient>();
```

Comment:
All the server calls to TCM are made through this interface `ITestResultsServer` can you also follow the same pattern here.

And have a error handling for that method so, we are not affecting the main flow with this.